llama_stack/providers/tests/conftest.py

```diff
@@ -148,6 +179,41 @@ def pytest_itemcollected(item):
         item.name = f"{item.name}[{marks}]"
 
 
+def pytest_collection_modifyitems(session, config, items):
+    if config.getoption("--config") is None:
+        return
+    file_name = config.getoption("--config")
+    config_file_path = Path(__file__).parent / file_name
+    if not config_file_path.exists():
+        raise ValueError(
+            f"Test config {file_name} was specified but not found. Please make sure it exists in the llama_stack/providers/tests directory."
+        )
+
+    required_tests = dict()
+    inference_providers = set()
+    with open(config_file_path, "r") as config_file:
+        test_config = yaml.safe_load(config_file)
+        for test in test_config["tests"]:
+            required_tests[Path(__file__).parent / test["path"]] = set(
+                test["functions"]
+            )
+        inference_providers = set(test_config["inference_fixtures"])
+
+    new_items, deselected_items = [], []
```

Comment:
we should make sure this can generalize properly for other tests too (we will have memory, safety and agents tests at least)